### ISSUE TITLE
Use async_forward_entry_setups instead of async_setup_platforms

### ISSUE THREAD

**Issue by ArveVM** [erlendsellie/priceanalyzer#25]
## Version of the custom_component

1.2.1 is the lates according to HACS (when looking at PriceAnalyzer) - and no update available

## Homeassistant version

Home Assistant 2023.2.1 (running Core)

## Configuration

```yaml

Add your logs here.

```

## Describe the bug
It's a warning that things will fail horribly next month  ;)

Got exact similar warning for Frigate,, where it was reported and corrected, see ref:
https://github.com/blakeblackshear/frigate-hass-integration/issues/421

## Debug log

```text

Log Details (WARNING)
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:77
First occurred: 11:36:59 AM (2 occurrences)
Last logged: 11:37:02 AM

Detected integration that called async_setup_platforms instead of awaiting async_forward_entry_setups; this will fail in version 2023.3. Please report issue to the custom integration author for priceanalyzer using this method at custom_components/priceanalyzer/__init__.py, line 254: hass.config_entries.async_setup_platforms(entry, PLATFORMS)

```

**Comment by erlendsellie** [comment]
Thank's for letting me know, and a very vell written Issue Arve.
Fixed in the latest release.
